iis2iclx_reg.c

```diff
@@ -6114,23 +6114,30 @@ int32_t iis2iclx_mlc_out_get(const stmdev_ctx_t *ctx, uint8_t *buff)
   *
   */
 int32_t iis2iclx_sh_read_data_raw_get(const stmdev_ctx_t *ctx,
-                                      iis2iclx_emb_sh_read_t *val)
+                                      iis2iclx_emb_sh_read_t *val, uint16_t len)
 {
   int32_t ret;
 
+  /* Check on registers SENSOR_HUB_X range */
+  if (len > 18)
+  {
+    ret = -1; /* (or -EINVAL) */

```

Comment:
rest is good!!

Reply:
Honestly, I prefer  `-EINVAL` but it needs to include `errno.h` that is actually a standard but unfortunately it is not included in these drivers. 

I don't like to introduce the error code `-1` because it is undefined and can be in contrast with other error definition but without `errno.h` I can only suggest to use ` -EINVAL` in a comment.

Reply:
OK, I still have some doubts, but it may make sense. Anyway I would at least remove the word "or", i.e. like this:

```ret = -1; /* -EINVAL */```

I don't like he word "or" because it seems a user choice which is not. It is more an indication of the meaning.
Ah! Another possibility would be to use:

```ret = -22; /* -EINVAL */```

because it should be a standard value.

Reply:
I would use the former because it is simpler.